src/interface/block_chain.cpp

```diff
@@ -293,6 +299,251 @@ void block_chain::push(transaction_const_ptr tx, dispatcher&,
+                    txs_to_remove_from_list--;
+                }
+                // Add the new transaction ordered by benefit
+                insert_to_chosen_list(tx, benefit, tx_size, tx_sigops);
+            }
+        } else {
+            // Since there still space to add the transaction
+            // Add the new transaction ordered by benefit
+            insert_to_chosen_list(tx, benefit, tx_size, tx_sigops);
+        }
+    }
+
+    return true;
+} 
+
+std::vector<block_chain::tx_benefit> block_chain::get_gbt_tx_list() const{
+    if (stopped()) {
+        return std::vector<block_chain::tx_benefit>();
+    }
+
+    if (!gbt_ready_) {
+        return std::vector<block_chain::tx_benefit>();
+    }
+
+    std::lock_guard<std::mutex> lock(*gbt_mutex_);
+    std::vector<tx_benefit> txs_chosen_list{ std::begin(chosen_unconfirmed_), std::end(chosen_unconfirmed_) };
+
+    return txs_chosen_list;
+}
+
+
+//When a new block arrives, we need to check every transaction on the chosen list
+//to see if it was mined; and remove it if it was.
+//using tx_benefit = std::tuple<double /*benefit*/, size_t /*tx_sigops*/, size_t /*tx_size*/, size_t /*tx_fees*/, libbitcoin::data_chunk /*tx_hex*/, libbitcoin::hash_digest /*tx_hash */> ;
+bool block_chain::remove_mined_txs_from_chosen_list(block_const_ptr blk){
+
+    gbt_ready_= false;
+    std::lock_guard<std::mutex> lock(*gbt_mutex_);
+    for(auto const& tx : blk->transactions()){
+        //erase transactions by hash
+        auto it = std::find_if (chosen_unconfirmed_.begin(), chosen_unconfirmed_.end(),
+            [&](const tx_benefit& tx_chosen){
+                return tx_chosen.tx_id == tx.hash();
+              });
+        if(it != chosen_unconfirmed_.end()){
+          //If transaction was deleted by hash, remove spended outputs
+          remove_spend(tx.hash());
+          chosen_unconfirmed_.erase(it);
+        } else {
+            //If the tx mined hash wasnt on our chosen list
+            //There may be another transaction wich uses the same prev output as this tx. (double spend attempt) 
+            //This should be extremely rare
+            libbitcoin::message::transaction msg_tx {tx};
+            transaction_const_ptr msg_ptr = std::make_shared<const libbitcoin::message::transaction>(msg_tx);
+            for(auto const& conflict_hash : get_double_spend_chosen_list(msg_ptr)){
+                auto conflict = std::find_if (chosen_unconfirmed_.begin(), chosen_unconfirmed_.end(),
+                [&](const tx_benefit& tx_chosen){
+                    return (tx_chosen.tx_id == conflict_hash);
+                  });
+                if(conflict != chosen_unconfirmed_.end()){
+                    chosen_unconfirmed_.erase(conflict);
+                    remove_spend(conflict_hash); //Do it on get_souble_spend_mempool
+                }
+            }
+        }
+    }
+
+    chosen_size_ = 0;
+    chosen_sigops_ = 0;
+    for(auto tx : chosen_unconfirmed_ ){
```

Comment:
this will copy the `tx`, use `auto&` or `auto const&` (unless `tx` is something lightweight)